### PR TITLE
Rebase the two stranded freshness runs onto main and land their re-dates

### PR DESCRIPTION
**Pull request by ccerv1**
# land-stranded-freshness — pass

**Goal.** Two freshness runs did real work and never landed. **37 axes across 37 products** would move

## Success criteria
- [ ] Both branches are rebased onto current `main` and land as one PR or two; say which and why
- [ ] **No prose line from #620 is reverted.** Diff the score notes against `main` and show the count is zero, or name every exception and why
- [ ] **The invariant holds**: a check, in the PR or as a test, showing that every axis whose `last_verified` advanced also has its sources' `accessed` advanced. Quote its output
- [ ] `uv run python -m build.check_freshness` before and after, both quoted. The median and oldest should improve; say by how much rather than that they improved
- [ ] No axis is re-verified afresh. This is a rebase of work already done, not a new run — inventing a confirmation here would be fabricating evidence
- [ ] `uv run python -m build.validate` reports 0 errors
- [ ] `uv run python -m build.check_payload` reports ok
- [ ] `uv run pytest -q -n auto -m "not serial and not network"` passes, count quoted
- [ ] PR title names the change

## Outcome
`pass` after 1 round(s). Branch `loop/land-stranded-freshness-2026-09-20-a1`.

## Final verdict
```json
{
 "verdict": "pass",
 "blocking": [],
 "non_blocking": [],
 "settled_now": [
  "Accepted 33 date advances across 37 products; four products only refreshed source reads. No dates should be invented.",
  "Verified zero #620 prose reversions and zero date/digest provenance mismatches. Both original runs pass the re-dating comparison; the resulting tree passes the existing evidence invariant.",
  "Accepted unchanged corpus median 38d and oldest 43d: improving untouched axes requires prohibited fresh verification. The affected axes improve instead.",
  "Accepted one combined PR because four products overlap, with #597 taking precedence.",
  "Validation reports '0 error(s), 2 warning(s)'; payload reports '763 products, 421 organizations, ok'. Full-suite result relies on the builder's '2256 passed, 2 skipped'; reviewer rerun was blocked by the read-only sandbox."
 ],
 "escalation": null
}
```

## Settled ground this run added (7)
- Prose from `main`, dates and evidence from the PR, #597 over #595 on overlap.
- This is a rebase. Re-running the verifier is out of scope, and inventing a confirmation is worse
- Accepted 33 date advances across 37 products; four products only refreshed source reads. No dates should be invented.
- Verified zero #620 prose reversions and zero date/digest provenance mismatches. Both original runs pass the re-dating comparison; the resulting tree passes the existing evidence invariant.
- Accepted unchanged corpus median 38d and oldest 43d: improving untouched axes requires prohibited fresh verification. The affected axes improve instead.
- Accepted one combined PR because four products overlap, with #597 taking precedence.
- Validation reports '0 error(s), 2 warning(s)'; payload reports '763 products, 421 organizations, ok'. Full-suite result relies on the builder's '2256 passed, 2 skipped'; reviewer rerun was blocked by the read-only sandbox.

Later runs of this job inherit `/workspace/GitHub/personal/agent-loops/jobs/settled/land-stranded-freshness.json`; prune it there.

_Opened by build-review-loop; builder notes in the first comment._

### CODE REVIEW

**Comment by ccerv1** [comment]
## What I did

Landed both stranded freshness runs as **one PR, three commits**, on
`loop/land-stranded-freshness-2026-09-20-a1`:

- `ca6a3f74` — cherry-pick of `reverify/2026-09-15` (PR #595, the rolling deterministic leg), 20 products
- `cbad163f` — cherry-pick of `reverify-agent/2026-09-16` (PR #597, the agent leg), 21 products
- `0ca82141` — `build/check_redate.py` + `tests/test_check_redate.py`, the invariant gate

**Proposed PR title:** `Land the two stranded re-verification runs, and gate a re-dating against its reads`

- files touched: 37 files under `sources/scores/` (20 from the deterministic leg, 21 from the
  agent leg, 4 in both: `amazon-bedrock-custom-models-fine-tuning`, `cerebras-inference`,
  `sambanova-cloud`, `tinker`); new `build/check_redate.py`; new `tests/test_check_redate.py`.
  Nothing else in the repo changed — no payload regeneration was needed.
- commands run:
  - `git fetch <real repo> refs/remotes/origin/reverify/2026-09-15 refs/remotes/origin/reverify-agent/2026-09-16` (read-only; the clone has no remotes)
  - `git cherry-pick -x` each stranded commit onto `main`, 3 then 9 conflicted files resolved
  - `uv run python -m build.check_freshness` (before and after)
  - `uv run python -m build.check_redate --base main`
  - `uv run python -m build.validate`
  - `uv run python -m build.check_payload`
  - `uv run pytest -q -n auto -m "not serial and not network"`

### One PR or two, and why

**One.** The two runs overlap on four products, and on every overlapping field the agent leg
(2026-09-16) supersedes the deterministic leg (2026-09-15) — same URL, later read, different
digest. Landing them as two PRs would mean the second one conflicts with the first on exactly
those four files and a reviewer resolves the same superseding by hand a second time. They stay
as two commits so each run's provenance and original message survive.

### No #620 prose reverted: the count is zero

Both branches were cut *before* `e69d401b` "Write the published prose for the reader, not the
score auditor (#620)", which rewrote `note` and some `shows` text across the corpus. 36 of the
37 files this branch touches were also touched by #620, so this was the real risk.

Check: for every leaf #620 changed in those 36 files, compare `e69d401b`'s value to this tree's.

```
files touched by both #620 and this branch: 36
REVERTED #620 leaves: 0
```

Classified the whole diff against `main` by field as well:

```
changed leaves by field:
  accessed: 80
  content_sha256: 43
  last_verified: 33
  detail: 1
  note: 1
  raw: 1
  shows: 1
prose-field changes: 4
```

All four prose changes are in `sources/scores/snowflake-cortex-fine-tuning.yaml`, which **#620
did not touch** (#620 touched `sources/products/snowflake-cortex-fine-tuning.yaml`, a different
file). They are the agent leg's substantive finding, not a rewording: Snowflake's base-model
allowlist narrowed from six models to one (`llama3.1-8b`), which the run recorded in
`openness.components.base-models.detail`, `openness.raw`, `openness.note` and the source's
`shows`. Dropping them would have dropped the finding. Those are the only exceptions, and they
revert nothing.

### The invariant

`build/check_redate.py`: every axis whose `last_verified` advanced must (a) have a source
accessed on the new date and (b) have its newest `accessed` advanced too.

```
$ uv run python -m build.check_redate --base main
33 axis re-dating(s) across 33 product(s) since main
  alpacaeval.openness: 2026-08-13 -> 2026-09-15
  amazon-bedrock-custom-models-fine-tuning.openness: 2026-09-10 -> 2026-09-16
  anyscale-fine-tuning.openness: 2026-09-09 -> 2026-09-16
  apple-core-ml-runtime.openness: 2026-09-09 -> 2026-09-16
  azure-openai-fine-tuning.openness: 2026-09-09 -> 2026-09-16
  bigcode-dataset.openness: 2026-08-13 -> 2026-09-15
  cerebras-inference.openness: 2026-09-09 -> 2026-09-16
  compar-ia.openness: 2026-08-13 -> 2026-09-15
  daytona-sandbox.openness: 2026-08-12 -> 2026-09-15
  fireworks-fine-tuning.openness: 2026-09-09 -> 2026-09-16
  fireworks-inference.openness: 2026-09-09 -> 2026-09-16
  google-cloud-tpu-inference.openness: 2026-09-09 -> 2026-09-16
  lamini.openness: 2026-08-09 -> 2026-09-15
  livebench.openness: 2026-08-12 -> 2026-09-15
  llama-cpp.openness: 2026-08-09 -> 2026-09-15
  llamafirewall.openness: 2026-08-12 -> 2026-09-15
  megatron-lm.openness: 2026-09-09 -> 2026-09-16
  mistral-fine-tuning-api.openness: 2026-09-09 -> 2026-09-16
  mmmu.openness: 2026-08-13 -> 2026-09-15
  mosaic-ai-model-training.openness: 2026-09-09 -> 2026-09-16
  mt-bench.openness: 2026-08-12 -> 2026-09-15
  openclaw.openness: 2026-08-13 -> 2026-09-15
  openhands.openness: 2026-08-12 -> 2026-09-15
  peft.openness: 2026-09-09 -> 2026-09-16
  sambanova-cloud.openness: 2026-09-09 -> 2026-09-16
  snowflake-cortex-fine-tuning.openness: 2026-09-09 -> 2026-09-16
  swift.openness: 2026-09-09 -> 2026-09-16
  text-generation-inference.openness: 2026-08-11 -> 2026-09-15
  tinker.openness: 2026-09-09 -> 2026-09-16
  trl.openness: 2026-09-09 -> 2026-09-16
  ungoliant.openness: 2026-08-13 -> 2026-09-15
  vertex-ai-tuning.openness: 2026-08-09 -> 2026-09-15
  vllm.openness: 2026-08-09 -> 2026-09-15

every re-dated axis was read on the date it now claims, from a fresher source
```

**33 axes, not 37.** The job brief said 37; 37 is the count of *product files* the two runs
touch. Four of them — `ds4`, `llama-factory`, `max`, `openai-fine-tuning-api` — had their
sources re-read (`accessed` and in three cases `content_sha256` moved) without the agent leg
re-dating the axis, which matches its own message: "18 products, 13 re-dated". Their
`last_verified` was left where the run left it. I did not fix the date to match the read; that
direction is correct behavior under the freshness rule, which never backfills `last_verified`
from `accessed`.

### check_freshness, before and after

Before (`main`, e6759665):

```
finetuning_code                   96      40d      43d  trl.adoption
inference_code                   117      20d      42d  llama-cpp.openness
2289 axes | median 38d | oldest 43d | newest 3d
```

After:

```
finetuning_code                   96      38d      43d  trl.adoption
inference_code                   117      20d      42d  llama-cpp.adoption
2289 axes | median 38d | oldest 43d | newest 3d
```

**The corpus median and oldest do not move, and I am not going to claim they do.** Median stays
38d and oldest stays 43d, because 33 axes are 1.4% of 2289 and the oldest axis
(`trl.adoption`, 43d) is an adoption axis that neither run touched — both legs re-verified
openness only. What moved:

- the 33 re-dated axes: median age **11d → 4d**, mean **24.8d → 4.5d**, oldest **42d → 5d**
- corpus mean age: **28.63d → 28.34d**; 669 axis-days of staleness removed
- `finetuning_code` median: **40d → 38d** (the only category median to move)
- `inference_code`'s oldest axis changes hands from `llama-cpp.openness` to `llama-cpp.adoption`
  at the same 42d — the openness axis moved out from under it

### Gates

```
$ uv run python -m build.validate
0 error(s), 2 warning(s)          # both pre-existing model_families pattern-overlap warnings

$ uv run python -m build.check_payload
check_payload: 763 products, 421 organizations, ok

$ uv run pytest -q -n auto -m "not serial and not network"
2256 passed, 2 skipped in 353.39s (0:05:53)
```

## Why

- decisions:
  - **Cherry-pick, not `git rebase` or a hand-transplant of dates.** A cherry-pick is a
    three-way merge against the real merge-base, so a line neither run touched keeps `main`'s
    text by construction. That is what makes "zero #620 lines reverted" a property of the
    method rather than of my care.
  - **Conflict rule: `main`'s prose, the run's dates and digests.** Every conflict was one of
    two shapes. (1) #620 rewrote a `note` or `shows` adjacent to a date the run moved — 6 files
    (`compar-ia`, `text-generation-inference`, `vllm`, `anyscale-fine-tuning`,
    `apple-core-ml-runtime`, `mistral-fine-tuning-api`, plus the `shows` cases in `megatron-lm`
    and `swift`). I checked base→branch for each before resolving: in all of them the run left
    the prose alone, so the conflict was pure adjacency and `main` wins the words. (2) The two
    runs re-read the same source — 4 files. The later read (09-16) wins both date and digest.
    Resolution was scripted over the conflict hunks so the rule is applied identically
    everywhere: keep `ours` lines, substitute `theirs` values for `last_verified` / `accessed` /
    `content_sha256`, asserting the key names line up.
  - **No load-and-dump of the YAML.** Nothing re-serialized a file. Every edit is a line
    substitution inside a conflict hunk, so the hand wrapping outside those hunks is untouched.
  - **The invariant as a module plus unit tests, not a one-off script.** `check_redate.py`
    mirrors `check_prose_diff.py` (its inverse: that one says a prose pass moves no date, this
    one says a date has a read behind it) and it is the gate that would have caught a bad merge
    of these two branches. The tests exercise the pure comparison, so they need no git state.
  - **No fetching.** Zero network calls. Every date and digest here comes from a run that
    already happened.
- alternatives rejected:
  - *Two PRs.* Rejected above: the four overlapping products would make the second PR conflict
    with the first.
  - *Deriving the value map and re-applying it onto `main` by hand.* It produces a smaller diff
    (it would drop the bot's incidental re-wrapping and its `establishes: [a, b]` → block-list
    reformat in the deterministic leg's files) but it silently discards anything I failed to
    enumerate — including the Snowflake finding, which is not a date at all. Provenance beats
    diff tidiness here.
  - *Advancing `last_verified` on the four products whose sources were re-read but whose axis
    the run did not re-date.* That is inventing a confirmation.

## Risks / unsure about

- The deterministic leg's commit carries incidental formatting churn inside the files it
  touched: re-wrapped `shows` scalars and `establishes: [a, b]` flow lists expanded to block
  lists. It is real diff noise (roughly the difference between 125 insertions and the ~40 lines
  of actual date and digest change in `ca6a3f74`). I left it, because it is what the bot
  committed and stripping it means hand-editing 20 files to no semantic end. Worth a reviewer
  knowing where the noise comes from.
- Four products (`ds4`, `llama-factory`, `max`, `openai-fine-tuning-api`) now carry a source
  read on 2026-09-16 under an axis dated 2026-09-09 or earlier. That is correct under the
  freshness rule and `check_redate` deliberately does not flag it, but it is the sort of thing a
  future reader may mistake for a bug. It is also a real signal that those four are due a
  re-verification pass.
- `snowflake-cortex-fine-tuning` is the one place a score's *meaning* changes here, not just its
  date. The allowlist narrowing is a genuine finding from the agent run and I carried it
  verbatim; I did not re-check the claim against Snowflake's page, because re-checking would be
  a fresh verification and the brief rules that out. Someone should sanity-read that one file.
- The invariant gate is not wired into CI. I did not add it to a workflow, because the
  refresh-chain cadence question (which leg runs it, and when) is not mine to settle in a rebase
  PR.

## Response to round N-1

| item (where) | fixed | pushed_back | escalate | evidence |
|---|---|---|---|---|
| (round 1 — nothing to respond to) | — | — | — | — |
